pkg/protocol/engine/commitmentfilter/accountsfilter/commitmentfilter.go

```diff
@@ -58,6 +64,36 @@ func (c *CommitmentFilter) ProcessPreFilteredBlock(block *blocks.Block) {
 }
 
 func (c *CommitmentFilter) evaluateBlock(block *blocks.Block) {
+	// Block issuing time monotonicity: a block's issuing time needs to be greater than its parents issuing time.
+	for _, parentID := range block.Parents() {
+		var parentIssuingTime time.Time
+
+		if parentID == iotago.EmptyBlockID {
```

Comment:
this is wrong. Genesis is not the `EmptyBlockID`. You can fetch the genesis blockID from the ProtocolParameters

Reply:
Removed this entire check. I realized that we actually do have the issuing time in the root blocks. Therefore, it is simpler and more performant to just use blocks from the block cache instead.